pkg/provisioningv2/rke2/planner/encryptionkeyrotation.go

```diff
@@ -290,59 +484,64 @@ func (p *Planner) encryptionKeyRotationUpdateControlPlanePhase(cp *rkev1.RKECont
 	return nil
 }
 
-// encryptionKeyRotationPhaseFromPeriodic will attempt to extract the current phase (secrets-encrypt status) from the
-// plan by parsing the periodic output, or the one time output. In most cases, the periodic output will be the only one
-// populated, however the leader requires deterministic status when entering the apply sequence, as clusters with few
-// nodes can complete the 'prepare' sequence before the leader has time to rerun the periodic status command, which
-// could potentially cause a duplicate apply.
-func encryptionKeyRotationPhaseFromPeriodic(plan *planEntry) (string, error) {
-	output, ok := plan.Plan.PeriodicOutput[SecretsEncryptStatusCommand]
+// encryptionKeyRotationSecretsEncryptStageFromPeriodic will attempt to extract the current stage (secrets-encrypt status) from the
+// plan by parsing the periodic output.
+func encryptionKeyRotationSecretsEncryptStageFromPeriodic(plan *planEntry) (string, error) {
+	output, ok := plan.Plan.PeriodicOutput[encryptionKeyRotationSecretsEncryptStatusCommand]
 	if !ok {
-		return "", ErrWaitingf("Could not extract current status from plan for [%s]: no output for status", plan.Machine.Name)
+		for _, pi := range plan.Plan.Plan.PeriodicInstructions {
+			if pi.Name == encryptionKeyRotationSecretsEncryptStatusCommand {
+				return "", ErrWaitingf("could not extract current status from plan for [%s]: no output for status", plan.Machine.Name)
```

Comment:
An example of returning `ErrWaiting` which will cause the `Ready` condition to have a reason of `Waiting`.